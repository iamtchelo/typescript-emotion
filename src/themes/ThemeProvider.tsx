import { ThemeProvider as Provider } from 'emotion-theming';
import * as React from 'react';
import { ComponentEnhancer, compose, setDisplayName ,withState } from 'recompose';
import primaryTheme from './primary';
import secondaryTheme from './secondary';

const themes = {
  primary: primaryTheme,
  secondary: secondaryTheme
};

export interface ThemeProviderProps {
  theme?: 'primary' | 'secondary';
  children?: React.SFC<any>;
};

export interface ThemeProviderDispatches {
  setTheme: (theme: 'primary' | 'secondary') => void;
};

export type EnhancedThemeProviderProps = ThemeProviderProps & ThemeProviderDispatches;

const ThemeProvider: React.SFC<EnhancedThemeProviderProps> = ({
  theme = 'primary',
  setTheme,
  children
}) => (
  <Provider theme={themes[theme]}>
    {children && children({ theme, setTheme })}
  </Provider>
);

const withTheme = compose(
  withState('theme', 'setTheme', ({ theme = 'primary' }: any) => theme)
);

const enhance = compose(
  setDisplayName('ThemeProvider'),
  withTheme
) as ComponentEnhancer<EnhancedThemeProviderProps, ThemeProviderProps>

export default enhance(ThemeProvider);
