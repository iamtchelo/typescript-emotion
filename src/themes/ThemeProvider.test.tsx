import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import * as React from 'react';
import ThemeProvider from './ThemeProvider';

describe('<ThemeProvider />', () => {
  it('should render with a default theme', () => {
    const tree = shallow(<ThemeProvider />);

    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render with a theme prop', () => {
    const tree = shallow(<ThemeProvider theme="secondary" />);

    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
