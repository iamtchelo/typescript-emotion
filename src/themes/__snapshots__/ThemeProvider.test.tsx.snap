// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ThemeProvider /> should render with a default theme 1`] = `
<ThemeProvider
  theme={
    Object {
      "button": Object {
        "background": "#FFFFFF",
        "backgroundActived": "#80a1c1",
        "backgroundHover": "#7593b0",
        "borderColor": "#80a1c1",
        "color": "#414141",
      },
      "card": Object {
        "titleColor": "#808080",
      },
      "colors": Object {
        "gray": "#414141",
        "grayDark": "#808080",
        "grayLight": "#E0E0E0",
        "primary": "#80a1c1",
        "primaryHover": "#7593b0",
        "white": "#FFFFFF",
      },
      "navbar": Object {
        "background": "#80a1c1",
        "titleColor": "#FFFFFF",
      },
    }
  }
/>
`;

exports[`<ThemeProvider /> should render with a theme prop 1`] = `
<ThemeProvider
  theme={
    Object {
      "button": Object {
        "background": "#FFFFFF",
        "backgroundActived": "#b5ca8d",
        "backgroundHover": "#a5b881",
        "borderColor": "#b5ca8d",
        "color": "#414141",
      },
      "card": Object {
        "titleColor": "#808080",
      },
      "colors": Object {
        "gray": "#414141",
        "grayDark": "#808080",
        "grayLight": "#E0E0E0",
        "primary": "#b5ca8d",
        "primaryHover": "#a5b881",
        "white": "#FFFFFF",
      },
      "navbar": Object {
        "background": "#b5ca8d",
        "titleColor": "#FFFFFF",
      },
    }
  }
/>
`;
